carma_wm/include/carma_wm/collision_detection.h

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2019 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include <cav_msgs/RoadwayObstacleList.h>
+#include <cav_msgs/RoadwayObstacle.h>
+#include <cav_msgs/VehicleSize.h>
+#include <cav_msgs/TrajectoryPlan.h>
+#include <geometry_msgs/Vector3.h>
+
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/point_xy.hpp>
+#include <boost/foreach.hpp>
+#include <vector>
+#include <boost/assign/std/vector.hpp>
+
+#include <iostream>
+#include <fstream>
+
+#include <carma_wm/Geometry.h>
+#include <tf2/LinearMath/Transform.h>
+#include <tf2_geometry_msgs/tf2_geometry_msgs.h>
+
+
+namespace carma_wm {
+
+    namespace collision_detection {
+
+        typedef boost::geometry::model::point<double, 2, boost::geometry::cs::cartesian> point_t;
+        typedef boost::geometry::model::polygon<point_t> polygon_t;
+
+        struct MovingObject {
+            polygon_t object_polygon;
+            geometry_msgs::Vector3 linear_velocity;
+            std::vector<polygon_t> future_polygons;
+        };
+
+        /*!
+        * Main class for the CollisionChecking interfacing.
+        */
+
+        class CollisionChecking {
```

Comment:
It looks like this class is stateless and has no private members. So it would be better to just implement the functions as free functions inside the collision_detection namespace.